File: frontend/src/types/learning-space.ts
export type SpaceStatus = 'active' | 'archived'

export interface LearningSpace {
  id: string
  name: string
  mode: 'exam'
  goal: string
  exam_date: string
  daily_minutes: number
  status: SpaceStatus
  created_at: string
  updated_at: string
}

export interface CreateSpaceInput {
  name: string
  mode: 'exam'
  goal: string
  exam_date: string
  daily_minutes: number
}

export interface SpaceSummary {
  material_count: number
  node_count: number
  mastered_node_count: number
  today_task_count: number
}

export interface SpaceDetail {
  space: LearningSpace
  summary: SpaceSummary
}

File: backend/internal/dto/learning_space.go
package dto

import "whatsnext/backend/internal/model"

type CreateSpaceInput struct {
	Name         string `json:"name"`
	Mode         string `json:"mode"`
	Goal         string `json:"goal"`
	ExamDate     string `json:"exam_date"`
	DailyMinutes int    `json:"daily_minutes"`
}
type UpdateSpaceInput struct {
	Name         *string `json:"name"`
	Goal         *string `json:"goal"`
	ExamDate     *string `json:"exam_date"`
	DailyMinutes *int    `json:"daily_minutes"`
	Status       *string `json:"status"`
}
type SpacePage struct {
	Items      []model.LearningSpace `json:"items"`
	NextCursor *string               `json:"next_cursor"`
}
type SpaceSummary struct {
	MaterialCount     int `json:"material_count"`
	NodeCount         int `json:"node_count"`
	MasteredNodeCount int `json:"mastered_node_count"`
	TodayTaskCount    int `json:"today_task_count"`
}
type SpaceDetail struct {
	Space   model.LearningSpace `json:"space"`
	Summary SpaceSummary        `json:"summary"`
}
